ISM43362Interface.cpp

```diff
@@ -544,3 +544,12 @@ void ISM43362Interface::event()
         }
     }
 }
+
+#if MBED_CONF_ISM43362_PROVIDE_DEFAULT
```

Comment:
Maybe this new macro can be checked at the beginning in order to avoid full file compilation ?

Reply:
Full file should be compiled anyway, applications can still directly create instance of this driver.

This would be in case application is doing:
```
wifi = WiFiInterface::get_default_instance();
int ret = wifi->connect(...);
```

which works after Mbed OS 5.9, but not on external drivers. This PR adds capability of this to external ISM driver as well.